File: GitNexus/gitnexus/test/fixtures/sample-code/simple.ts
export interface UserConfig {
  name: string;
  email: string;
  active: boolean;
}

export function validateUser(config: UserConfig): boolean {
  return config.name.length > 0 && config.email.includes('@');
}

export class UserService {
  private users: UserConfig[] = [];

  addUser(user: UserConfig): void {
    if (validateUser(user)) {
      this.users.push(user);
    }
  }

  getUser(name: string): UserConfig | undefined {
    return this.users.find(u => u.name === name);
  }
}

function internalHelper(): string {
  return 'helper';
}

File: GitNexus/gitnexus/test/fixtures/sample-code/simple.rs
pub fn public_function(x: i32) -> i32 {
    x + 1
}

fn private_function() -> &'static str {
    "private"
}

pub struct Config {
    pub name: String,
}

impl Config {
    pub fn new(name: &str) -> Self {
        Config { name: name.to_string() }
    }
}
